File: javascript/kindergarten-garden/kindergarten-garden.js
var DEFAULT_STUDENTS = [ 'Alice', 'Bob', 'Charlie', 'David', 'Eve',
		'Fred', 'Ginny', 'Harriet', 'Ileana', 'Joseph', 'Kincaid', 'Larry'];

var Garden = function(garden, students){
  /* A Kindergarden plant garden */
  students ? students.sort(): (students = DEFAULT_STUDENTS);
  garden = garden.split('\n').map(MakeRow);
};

function MakeRow(row) {
	/* Converts a row of plant letters to an array of plants */
}

function StudentsGarden(garden, position) {
	/* The plants owned by an individual student */

}

module.exports = Garden;

File: ruby/kindergarten-garden/kindergarten_garden.rb
class Garden

  def initialize(diagram, students = default_children)
    @pots = parse(diagram)
    @students = students.sort
    assign_pots
  end

  def default_children
    %w(Alice Bob Charlie David Eve Fred Ginny Harriet Ileana Joseph Kincaid Larry)
  end

end
